### ISSUE TITLE
TLS Client always gets disconnected after no connect timeout

### ISSUE THREAD

**Issue by michaelg9** [hivemq/hivemq-community-edition#149]
## Expected behavior
The client has some time (no-connect-idle-timeout) after the TCP handshake to send the CONNECT packet otherwise it gets disconnected. If the client sends a connect packet, then they shouldn't be disconnected after a successful authentication
## Actual behavior
A TLS client always gets disconnected after no-connect-idle-timeout

### Steps
Setup a broker with the latest version (4.3) and connect a subscriber and just let it wait until the no-connect-idle-timeout timer goes out.

### Cause
The issue is that TlsTcpChannelInitializer adds the listener for no connect idle timeout to the end of the pipeline, instead of after Message encoder, as the default behaviour is:
```java
    @Override
    protected void addNoConnectIdleHandler(@NotNull final Channel ch) {
        // No connect idle handler are added, as soon as the TLS handshake is done.
    }

    protected void addNoConnectIdleHandlerAfterTlsHandshake(@NotNull final Channel ch) {
        super.addNoConnectIdleHandler(ch);
    }

   @Override
    protected void addSpecialHandlers(@NotNull final Channel ch) throws SslException {

        final int handshakeTimeout = tlsTcpListener.getTls().getHandshakeTimeout();
        final IdleStateHandler idleStateHandler = new IdleStateHandler(handshakeTimeout, 0, 0, TimeUnit.MILLISECONDS);
        final NoTlsHandshakeIdleHandler noTlsHandshakeIdleHandler = new NoTlsHandshakeIdleHandler(eventLog);
        if (handshakeTimeout > 0) {
            ch.pipeline().addLast(NEW_CONNECTION_IDLE_HANDLER, idleStateHandler);
            ch.pipeline().addLast(NO_TLS_HANDSHAKE_IDLE_EVENT_HANDLER, noTlsHandshakeIdleHandler);
        }

        final Tls tls = tlsTcpListener.getTls();
        final SslHandler sslHandler = sslFactory.getSslHandler(ch, tls);
        sslHandler.handshakeFuture().addListener(future -> {
            if (handshakeTimeout > 0) {
                ch.pipeline().remove(idleStateHandler);
                ch.pipeline().remove(noTlsHandshakeIdleHandler);
            }
            addNoConnectIdleHandlerAfterTlsHandshake(ch);
        });

        new SslInitializer(sslHandler, tls, eventLog, sslParameterHandler).addHandlers(ch);
    }
```
Then the listener is at the end of the pipeline and the connect message doesn't make it there because CONNECT packets end in ConnectHandler, which doesn't forward them on to the pipeline. Thus, the channel read method of NoConnectIdleHandler never gets called to disable the timer and the client is disconnected as soon as the timer goes off.

### Potential bugfix
Instead of adding SslHandler and NoConnectIdleHandler as special handlers at the end of the pipeline after all MQTT specific handlers, add them in the same place as the plain TCP pipeline has the NoConnectIdleHandler. To do this, just overwrite addNoConnectIdleHandler and only add the SslHandler, since the NoConnectIdleHandler is dealt with by the parent class.

## Details
- Affected HiveMQ CE version(s): latest
- Used JVM version: 11

**Comment by SgtSilvio** [comment]
Hi @michaelg9 
thank you for being an active community member.
I can confirm that this is a bug in HiveMQ CE 2020.1
We already found this behavior and will fix it soon.
We will update this ticket when a fix is available.

**Comment by SgtSilvio** [comment]
Hi @michaelg9 
The issue is fixed in the [2020.2](https://github.com/hivemq/hivemq-community-edition/releases/tag/2020.2) release
